pkg/kubelet/kubelet.go

```diff
@@ -3567,7 +3579,18 @@ func (kl *Kubelet) ListenAndServeReadOnly(address net.IP, port uint) {
 	server.ListenAndServeKubeletReadOnlyServer(kl, kl.resourceAnalyzer, address, port, kl.containerRuntime)
 }
 
-// Filter out events that are not worthy of pod syncing
+// deleteContainersInPod issues container deletion requests
+func (kl *Kubelet) deleteContainersInPod(candidates ContainerStatusbyCreated) {
+	for _, candidate := range candidates {
+		select {
+		case kl.containerDeletionWorker <- candidate.ID:
```

Comment:
It's non-blocking (fire and continue) to protect the syncloop. If the channel is full, the default case is triggered.